power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/serialization/deserializer.hpp

```diff
@@ -44,18 +44,21 @@ using nlohmann::json;
 // visitor for json conversion
 struct JsonMapArrayData {
     size_t size{};
-    msgpack::sbuffer buffer{};
```

Comment:
why use a `shared_ptr` here? It does not make any sense?

Reply:
`msgpack::sbuffer` has copy constructor `delete` but the copy constructor is a prerequisite for `std::stack`. This makes it copyable. If you have any better ideas please lmk

Reply:
cfr discussion: reverted and we'll see what sonar cloud thinks of it